go/vt/vtgate/vstream_manager.go

```diff
@@ -214,6 +221,58 @@ func (vs *vstream) stream(ctx context.Context) error {
 	return vs.err
 }
 
+func (vs *vstream) sendEvents(ctx context.Context) {
+	var heartbeatDuration time.Duration
+	if vs.heartbeatInterval == 0 {
+		heartbeatDuration = time.Duration(math.MaxInt64) * time.Nanosecond // max time.Duration => 290 years
+	} else {
+		heartbeatDuration = time.Duration(vs.heartbeatInterval) * time.Second
+	}
+
+	timer := time.NewTicker(heartbeatDuration)
+	defer timer.Stop()
```

Comment:
So, I'm a bit spooked about keeping a timer with `MaxInt64` for the case where we're not supposed to heartbeat. I don't know how well the runtime deals with such large periods and whether it could overflow or lead to weird behavior.

What do you think about this refactoring?

```suggestion
	var heartbeat <-chan time.Time
	var resetHeartbeat func()
	
	if vs.heartbeatInterval == 0 {
		heartbeat = make(chan time.Time)
		resetHeartbeat = func() {}
	} else {
		d := time.Duration(vs.heartbeatInterval) * time.Second
		timer := time.NewTicker(d)
		defer timer.Stop()
		
		heartbeat = timer.C
		resetHeartbeat = func() { timer.Reset(d) }
	}

```

Reply:
Elegant solution, thanks! Implemented. I was not so crazy about that MaxInt solution as well!